engine/src/test/java/org/camunda/bpm/engine/test/bpmn/tasklistener/TaskListenerEventLifecycleTest.java

```diff
@@ -0,0 +1,586 @@
+
+    assertThat(orderedEvents.size()).isEqualTo(4);
+    assertThat(orderedEvents).containsExactly(TaskListener.EVENTNAME_CREATE,
+                                              TaskListener.EVENTNAME_UPDATE,
+                                              TaskListener.EVENTNAME_ASSIGNMENT,
+                                              TaskListener.EVENTNAME_COMPLETE);
+  }
+
+  @Test
+  public void testNoUpdateEventAfterCreateTaskListenerUpdatesProperties() {
+    // given
+    BpmnModelInstance process = createModelWithTaskEventsRecorderOnAssignedUserTask(TRACKED_EVENTS,
+                                                                                  null,
+                                                                                  TaskListener.EVENTNAME_CREATE,
+                                                                                  ModifyingTaskListener.class);
+    testRule.deploy(process);
+
+    // when
+    engineRule.getRuntimeService().startProcessInstanceByKey("process");
+
+    // then
+    List<String> orderedEvents = RecorderTaskListener.getOrderedEvents();
+
+    assertThat(orderedEvents.size()).isEqualTo(2);
+    // ASSIGNMENT Event is fired, since the ModifyingTaskListener sets an assignee, and the
+    // ASSIGNMENT Event evaluation happens after the CREATE Event evaluation
+    assertThat(orderedEvents).containsExactly(TaskListener.EVENTNAME_CREATE,
+                                              TaskListener.EVENTNAME_ASSIGNMENT);
+  }
+
+  @Test
+  public void testNoUpdateEventAfterUpdateTaskListenerUpdatesProperties() {
+    // given
+    BpmnModelInstance process = createModelWithTaskEventsRecorderOnAssignedUserTask(TRACKED_EVENTS,
+                                                                                    null,
+                                                                                    TaskListener.EVENTNAME_UPDATE,
+                                                                                    ModifyingTaskListener.class);
+    testRule.deploy(process);
+    engineRule.getRuntimeService().startProcessInstanceByKey("process");
+    Task task = engineRule.getTaskService().createTaskQuery().singleResult();
+
+    // when
+    taskService.setPriority(task.getId(), 3000);
+
+    // then
+    // only the initial, first update event is expected
+    List<String> orderedEvents = RecorderTaskListener.getOrderedEvents();
+
+    assertThat(orderedEvents.size()).isEqualTo(3);
+    // ASSIGNMENT Event is fired, since the ModifyingTaskListener sets an assignee, and the
+    // ASSIGNMENT Event evaluation happens after the UPDATE Event evaluation
+    assertThat(orderedEvents).containsExactly(TaskListener.EVENTNAME_CREATE,
+                                              TaskListener.EVENTNAME_UPDATE,
+                                              TaskListener.EVENTNAME_ASSIGNMENT);
+  }
+
+  @Test
+  public void testNoUpdateEventAfterAssignmentTaskListenerUpdatesProperties() {
```

Comment:
for new test classes we should try to stay consistent with the test method naming and stick to "shouldXYZ"
```suggestion
  public void shouldNotFireUpdateEventAfterAssignmentTaskListenerUpdatesProperties() {
```